### PR TITLE
Temporal reduction all iterator

### PR DESCRIPTION
**Pull request by amandasd**
A new class called all_iterator is added to calendar_util and C++ version of temporal_reduction makes use of it.

### CODE REVIEW

File: data/teca_calendar_util.h [hunk 1]
```diff
@@ -402,6 +402,38 @@ class TECA_EXPORT n_steps_iterator : public interval_iterator
...
+     * @param[in] last_step the last step to include in the series or -1 to use all
+     * @returns 0 if successfully initialized
+     */
+    int initialize(const const_p_teca_variant_array &t,
+        const std::string &units, const std::string &calendar,
+        long first_step, long last_step) override;
+
+    using interval_iterator::initialize;
+
+    /** return a pair of time steps bracketing the current season.
```
[the remark this answers is not in the document]
Comment (amandasd) [reply data/teca_calendar_util.h:428]: > The comment here seems to be taken from the season iterator

All the other iterator (year, month, day, n_steps) have the same description. Should we change all of them?

[the remark this answers is not in the document]
Comment (burlen) [reply data/teca_calendar_util.h]: yes, and since it is my mistake I just added a patch to fix it.

**Review by burlen — APPROVED** [review]
looks good

File: data/teca_calendar_util.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (amandasd) [reply data/teca_calendar_util.h]: Thanks, Burlen!

**Pull request merged**